File: projects/android/Sample/app/src/main/java/com/ezored/sample/observers/AppLifecycleObserver.java
package com.ezored.sample.observers;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;
import com.ezored.sample.app.Application;

public class AppLifecycleObserver implements LifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    public void onMoveToForeground() {
        Application.getInstance().onMoveToForeground();
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    public void onMoveToBackground() {
        Application.getInstance().onMoveToBackground();
    }

}

File: projects/android/Sample/app/src/main/java/com/ezored/sample/observer/AppLifecycleObserver.kt
package com.ezored.sample.observer

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import com.ezored.sample.app.Application

class AppLifecycleObserver : LifecycleObserver {

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onMoveToForeground() {
        Application.instance.onMoveToForeground()
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onMoveToBackground() {
        Application.instance.onMoveToBackground()
    }
}
